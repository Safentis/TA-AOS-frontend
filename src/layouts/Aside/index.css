.aside {
  background-color: var(--main-dk-grey-color);
  border-right: 0.5rem solid var(--main-white-color);
}

.aside__inner {
  padding: 3rem 1rem;
}

.nav {
  margin-top: 4.75rem;
}

.nav__item {
  align-items: center;
  display: flex;
  text-align: center;
}

.nav__item:nth-of-type(2) {
  margin-top: 1.5rem;
}

.nav__list {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.nav__link {
  transition: 0.3s;
}

.nav__icon {
  font-size: 1.6rem;
  margin-left: 1rem;
}

.nav__link {
  color: var(--main-white-color);
  font-size: 1.5rem;
  font-weight: 300;
}

.nav__link:hover {
  color: var(--main-dark-blue-color);
}

.nav__link_active {
  color: var(--main-dark-blue-color) !important;
}

.nav__icon_active {
  color: var(--main-purple-color) !important;
}
