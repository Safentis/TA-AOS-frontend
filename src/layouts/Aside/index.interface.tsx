import { IconDefinition } from '@fortawesome/fontawesome-common-types';

export interface Props {
  className?: string;
}

export interface Link {
  link: string;
  to: string;
  icon: IconDefinition;
}
