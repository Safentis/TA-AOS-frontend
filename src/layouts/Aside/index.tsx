import React, { FC } from 'react';
import { Link, useLocation } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHome, faUser } from '@fortawesome/free-solid-svg-icons';

import './index.css';
import { Props, Link as LinkInterface } from './index.interface';
import { Pathes } from '../../utils/pathes';

const Aside: FC<Props> = ({ className = '' }): React.ReactElement => {
  const { pathname } = useLocation();

  //* Links array, that is template for components
  const links: LinkInterface[] = [
    { link: 'Home', to: Pathes.PATH_MAIN, icon: faHome },
    { link: 'About', to: Pathes.PATH_ABOUT, icon: faUser },
  ];

  // prettier-ignore
  const content = links.map(
    ({ link, to, icon }: LinkInterface, index: number) => (
      <li className="nav__item" key={index}>
        <Link className={`nav__link ${pathname === to ? "nav__link_active" : ""}`} to={to}>
          {link}
          <FontAwesomeIcon 
            className={`nav__icon ${pathname === to ? "nav__icon_active" : ""}`} 
            icon={icon} 
          />
        </Link>
      </li>
    ),
  );

  return (
    <aside className={`aside ${className}`}>
      <div className="aside__inner">
        <nav className="nav aside__nav">
          <ul className="nav__list">{content}</ul>
        </nav>
      </div>
    </aside>
  );
};

export default Aside;
