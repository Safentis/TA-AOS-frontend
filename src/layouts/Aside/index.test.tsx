import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Aside from './index';

import { Props } from './index.interface';

const setUp = (props: Props) => {
  return shallow(<Aside {...props} />);
};

describe('<Aside />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    className: 'custom-style',
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has 1 aside root element', () => {
      expect(component.find('aside')).toBeTruthy();
    });

    it('has 2 Link elements', () => {
      expect(component.find('Link').length === 2).toBeTruthy();
    });

    it('has 2 FontAwesomeIcon elements', () => {
      expect(
        component.find('FontAwesomeIcon').length === 2,
      ).toBeTruthy();
    });
  });
});
