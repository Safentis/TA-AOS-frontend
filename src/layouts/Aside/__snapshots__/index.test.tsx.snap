// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Aside /> component does match to the snapshot 1`] = `
<aside
  className="aside custom-style"
>
  <div
    className="aside__inner"
  >
    <nav
      className="nav aside__nav"
    >
      <ul
        className="nav__list"
      >
        <li
          className="nav__item"
          key="0"
        >
          <Link
            className="nav__link "
            to="/"
          >
            Home
            <FontAwesomeIcon
              border={false}
              className="nav__icon "
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f015",
                    "M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z",
                  ],
                  "iconName": "home",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </Link>
        </li>
        <li
          className="nav__item"
          key="1"
        >
          <Link
            className="nav__link "
            to="/about"
          >
            About
            <FontAwesomeIcon
              border={false}
              className="nav__icon "
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f007",
                    "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
                  ],
                  "iconName": "user",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </Link>
        </li>
      </ul>
    </nav>
  </div>
</aside>
`;
